Applications/WebPortal/WebApp/library/source/pages/account_page/compliance_page/security_input.tsx

```diff
@@ -0,0 +1,762 @@
+            </div>);
+        }
+      } else {
+        if(this.props.displaySize === DisplaySize.SMALL) {
+          return (
+            <div style={iconWrapperStyle}>
+              <img height={imageSize} width={imageSize}
+                src={SecurityInput.PATH + 'remove-grey.svg'}/>
+            </div>);
+        } else {
+          return (
+            <div style={iconWrapperStyle}
+                onClick={this.removeEntry.bind(this)}>
+              <img height={imageSize} width={imageSize}
+                src={SecurityInput.PATH + 'remove-grey.svg'}/>
+              <div style={SecurityInput.STYLE.iconLabelReadonly}>
+                {SecurityInput.REMOVE_TEXT}
+              </div>
+            </div>);
+        }
+      }
+    })();
+    const confirmationButton = (() => {
+      if(this.props.readonly) {
+        return (
+          <button className={css(SecurityInput.EXTRA_STYLE.button)}
+              onClick={this.toggleEditing}>
+            {SecurityInput.CONFIRM_TEXT}
+          </button>);
+      } else {
+        return (
+          <button className={css(SecurityInput.EXTRA_STYLE.button)}
+            onClick={this.onSubmitChange}>
+            {SecurityInput.SUBMIT_CHANGES_TEXT}
+          </button>);
+      }
+    })();
+    let displayValue  = '';
+    const entries = [];
+    const maxLength = (() => {
+      if(this.props.value.length > this.state.localValue.length) {
+        return this.props.value.length;
+      }else {
+        return this.state.localValue.length;
+      }
+    })();
+    for(let i = 0; i < maxLength; ++i) {
+      if(i < this.state.localValue.length) {
+        const sec = this.state.localValue[i].value as Nexus.Security;
+        if(this.props.readonly) {
+          entries.push(
+            <div style={SecurityInput.STYLE.scrollBoxEntryReadonly}>
+              {sec.symbol.toString()}
+            </div>);
+        } else if(this.state.selection === i) {
+          entries.push(
+            <div style={SecurityInput.STYLE.scrollBoxEntrySelected}
+                onClick={this.selectEntry.bind(this, i)}>
+              {sec.symbol.toString()}
+            </div>);
+        } else {
+          entries.push(
+            <div style={SecurityInput.STYLE.scrollBoxEntry}
+                onClick={this.selectEntry.bind(this, i)}>
+              {sec.symbol.toString()}
+            </div>);
+        }
+      }
+      if(i < this.props.value.length) {
+        const sec = this.props.value[i].value as Nexus.Security;
+        displayValue = displayValue.concat(sec.symbol.toString());
+        if(i >= 0 && i < this.props.value.length - 1 && 
+            this.props.value.length > 1) {
+          displayValue = displayValue.concat(', ');
+        }
+      }
+    }
+    return(
+      <div>
+        <input
+          style={SecurityInput.STYLE.textBox}
+          className={css(SecurityInput.EXTRA_STYLE.effects)}
+          value={displayValue}
+          onFocus={this.toggleEditing.bind(this)}
```

Comment:
These kinds of binds should be done in the constructor to avoid having to repeatedly create closures on every render.